conans/client/tools/win.py

```diff
@@ -635,7 +635,7 @@ def run_in_windows_bash(conanfile, bashcmd, cwd=None, subsystem=None, msys_mingw
         It requires to have MSYS2, CYGWIN, or WSL
     """
     env = env or {}
-    if platform.system() != "Windows":
```

Comment:
Cygwin also returns "Windows" here, could be only Msys will be different

Reply:
didn't check Cygwin (may check tomorrow), msys returns a different value for sure

Reply:
The fix seems fine anyway, it was just an observation.

Reply:
for the completeness.
Cygwin:
```
sse4@DESKTOP-LAD63V0 ~
$ python -c "import platform; print(platform.system())"
CYGWIN_NT-10.0-18363
```
MSYS2:
```
sse4@DESKTOP-LAD63V0 MSYS ~
$ python -c "import platform; print(platform.system())"
MSYS_NT-10.0-18363
```
MinGW32 (MSYS2):
```
sse4@DESKTOP-LAD63V0 MINGW32 ~
$ python -c "import platform; print(platform.system())"
MINGW32_NT-10.0-18363
```
MinGW64 (MSYS2):
```
sse4@DESKTOP-LAD63V0 MINGW64 ~
$ python -c "import platform; print(platform.system())"
MINGW64_NT-10.0-18363
```

Reply:
Curiously, it seems it can change, I am not sure why:

Cygwin:
``` ~
$ python -c "import platform; print(platform.system())"
Windows
```

In any case, doesn't seem relevant, if it can change in other computers, need to take it into account.

Reply:
this is probably not a python from a Cygwin itself. run `which python` to check.
if it's `/usr/bin/python`, then it's cygwin's one.
if it's something like `/cygdrive/c/users...`, then it's installed in the main Windows system.